From eea7bba746cb3a9c06e606a2cecb413d299d9788 Mon Sep 17 00:00:00 2001
From: Zhao Qiang <qiang.zhao@nxp.com>
Date: Wed, 1 Aug 2018 17:08:32 +0800
Subject: [PATCH 070/107] net: fman: Support both new and legacy FMan Compatibles

Recently  the FMan Port and MAC compatibles were changed.
This patch aligns the FMan Port and MAC compatibles
to the new FMan device tree binding document.
The FMan device tree binding document can be found in the Linux
kernel:
./Documentation/devicetree/bindings/net/fsl-fman.txt

This patch doesn't affect legacy compatibles support.

Signed-off-by: Zhao Qiang <qiang.zhao@nxp.com>
---
 drivers/net/fm/init.c |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/drivers/net/fm/init.c b/drivers/net/fm/init.c
index 71c3abe..48344f0 100644
--- a/drivers/net/fm/init.c
+++ b/drivers/net/fm/init.c
@@ -329,7 +329,8 @@ void fdt_fixup_fman_ethernet(void *blob)
 				ft_fixup_port(blob, &fm_info[i],
 					      "fsl,fman-1g-mac");
 		} else {
-			if (ft_fixup_port(blob, &fm_info[i], "fsl,fman-tgec"))
+			if (ft_fixup_port(blob, &fm_info[i], "fsl,fman-tgec") &&
+			    ft_fixup_port(blob, &fm_info[i], "fsl,fman-xgec"))
 				ft_fixup_port(blob, &fm_info[i],
 					      "fsl,fman-10g-mac");
 		}
-- 
1.7.1

